<?xml version="1.0" encoding="utf-8"?>
<section xmlns="http://docbook.org/ns/docbook" version="5.0" xml:id="zend.exception.using"><info><title>Utilizzo delle eccezioni</title></info>

    

    <para>
        Tutte le eccezioni lanciate dalle classi del Framework Zend dovrebbero estendere la classe base Zend_Exception.
    </para>

    <example xml:id="zend.exception.using.example"><info><title>Gestione di un'eccezione</title></info>
        
        <para>
            Il codice seguente dimostra come gestire un'eccezione lanciata all'interno di una classe del Framework Zend:
        </para>
        <programlisting role="php"><![CDATA[
try {
    // La chiamata a Zend_Loader::loadClass() con una classe non esistente
    // causa il lancio di un'eccezione in Zend_Loader
    Zend_Loader::loadClass('classenonesistente');
} catch (Zend_Exception $e) {
    echo "Catturata eccezione: " . get_class($e) . "\n";
    echo "Messaggio: " . $e->getMessage() . "\n";
    // Altro codice necessario per gestire l'errore
}

        ]]></programlisting>
    </example>

    <para>
        The documentation for each Zend Framework
        component and class will contain specific information on which methods
        throw exceptions, the circumstances that cause an exception to be thrown,
        and the class of all exceptions that may be thrown.
    </para>

    <para>
        La documentazione di ogni componente e classe del Framework Zend
        contiene informazioni specifiche su quali metodi lanciano un'eccezione,
        le circostanze che possono causare l'errore e le classi per tutte
        le eccezioni che possono essere lanciate.
    </para>

</section>
